voyager/plugins/client-update/state-lens/evm/src/main.rs

```diff
@@ -0,0 +1,306 @@
+                                client_type: ClientType::new(ClientType::ETHEREUM),
+                                chain_id: self.l2_chain_id.clone(),
+                                counterparty_chain_id: self.l1_chain_id.clone(),
+                                update_from,
+                                update_to,
+                            })],
+                            [],
+                            AggregateMsgUpdateClientsFromOrderedHeaders {
+                                ibc_spec_id: IbcUnion::ID,
+                                chain_id: self.l1_chain_id.clone(),
+                                client_id: RawClientId::new(self.l1_client_id.clone()),
+                            },
+                        ),
+                        seq([
+                            call(WaitForTrustedHeight {
+                                chain_id: self.l1_chain_id.clone(),
+                                ibc_spec_id: IbcUnion::ID,
+                                client_id: RawClientId::new(self.l1_client_id),
+                                height: update_to,
+                            }),
+                            continuation,
+                        ]),
+                    ])),
+                }
+            }
+            ModuleCall::FetchUpdateAfterL1Update(FetchUpdateAfterL1Update {
+                counterparty_chain_id,
+                update_to,
+                ..
+            }) => {
+                let voy_client = ext.try_get::<VoyagerClient>()?;
+                let l1_latest_height = voy_client
+                    .query_latest_height(self.l1_chain_id.clone(), true)
+                    .await?;
+                let l0_client_meta = voy_client
+                    .client_meta::<IbcUnion>(
+                        counterparty_chain_id.clone(),
+                        QueryHeight::Latest,
+                        self.l0_client_id,
+                    )
+                    .await?;
+                let l2_consensus_state_proof = serde_json::from_value::<MerkleProof>(
+                    voy_client
+                        .query_ibc_proof(
+                            self.l1_chain_id.clone(),
+                            QueryHeight::Specific(l1_latest_height),
+                            ConsensusStatePath {
+                                client_id: self.l1_client_id,
+                                height: update_to.height(),
+                            },
+                        )
+                        .await
+                        .expect("big trouble")
+                        .proof,
+                )
+                .expect("impossible");
+                // Dispatch an update for the L1 on the destination, then dispatch the L2 update on the destination
+                Ok(conc([
+                    promise(
+                        [call(FetchUpdateHeaders {
+                            client_type: ClientType::new(ClientType::COMETBLS),
```

Comment:
i know we have union hardcoded right now, but just leaving a note that this can be fetched from the voyager client (`voyager_client.client_info()`)